src/main/java/com/philips/research/spdxbuilder/persistence/spdx/SpdxWriter.java

```diff
@@ -33,23 +29,23 @@
     private static final List<String> SUPPORTED_HASH_KEYS =
             List.of("SHA1", "SHA224", "SHA256", "SHA384", "SHA512", "MD2", "MD4", "MD5", "MD6");
 
-    private final File file;
+    private final OutputStream stream;
     private final Map<Package, SpdxRef> identifiers = new HashMap<>();
 
     private int nextId = 1;
 
-    public SpdxWriter(File file) {
-        this.file = file;
+    public SpdxWriter(OutputStream stream) {
+        this.stream = stream;
     }
 
     @Override
     public void process(BillOfMaterials bom) {
-        System.out.println("Writing SBOM to '" + file + "'");
-        try (final var doc = new TagValueDocument(new FileOutputStream(file))) {
+        try (final var doc = new TagValueDocument(this.stream)) {
             writeDocumentInformation(doc, bom);
             generatePackageIdentifiers(bom);
             writePackages(doc, bom);
             writeCustomLicenses(doc);
+            this.stream.close();
```

Comment:
https://stackoverflow.com/questions/37532375/whats-best-practice-to-close-resoures

We should implement the AutoClosable interface so it is compatible with try catch finally to autoclose the resources. Now if line 46 or 47 failes, 48 will never execute.